src/core/public/fatal_errors/fatal_errors_service.tsx

```diff
@@ -62,41 +57,36 @@ export interface FatalErrorsSetup {
 /** @interal */
 export class FatalErrorsService {
   private readonly errorInfo$ = new Rx.ReplaySubject<ErrorInfo>();
-  private i18n?: I18nSetup;
 
-  constructor(private params: FatalErrorsParams) {
+  constructor(private rootDomElement: HTMLElement, private stopCoreSystem: () => void) {}
+
+  public setup({ i18n, injectedMetadata: injectedMetadata }: Deps) {
     this.errorInfo$
       .pipe(
         first(),
-        tap(() => this.onFirstError())
+        tap(() => this.onFirstError(injectedMetadata, i18n))
       )
       .subscribe({
         error: error => {
           // eslint-disable-next-line no-console
-          console.error('Uncaught error in fatal error screen internals', error);
+          console.error('Uncaught error in fatal error service internals', error);
         },
       });
-  }
-
-  public add: FatalErrorsSetup['add'] = (error, source?) => {
-    const errorInfo = getErrorInfo(error, source);
-
-    this.errorInfo$.next(errorInfo);
 
-    if (error instanceof Error) {
-      // make stack traces clickable by putting whole error in the console
-      // eslint-disable-next-line no-console
-      console.error(error);
-    }
+    const fatalErrorsSetup: FatalErrorsSetup = {
+      add: (error, source) => {
```

Comment:
did you remove `?` for source accidentally? as I can see `getErrorInfo` still handles it as an optional parameter